runtime/src/main/java/io/novafoundation/nova/runtime/network/updaters/SingleChainUpdateSystem.kt

```diff
@@ -18,13 +20,15 @@ abstract class SingleChainUpdateSystem<A>(
 
     abstract fun getUpdaters(selectedAssetOption: SupportedAssetOption<A>): Collection<Updater<*>>
 
-    override fun start(): Flow<Updater.SideEffect> = singleAssetSharedState.selectedOption.flatMapLatest { selectedOption ->
+    private val updateFlow = singleAssetSharedState.selectedOption.flatMapLatest { selectedOption ->
         val chain = selectedOption.assetWithChain.chain
 
         val updaters = getUpdaters(selectedOption)
 
         runUpdaters(chain, updaters)
-    }.flowOn(Dispatchers.Default)
+    }.shareIn(CoroutineScope(Dispatchers.Default), replay = 1, started = SharingStarted.WhileSubscribed())
+
+    override fun start(): Flow<Updater.SideEffect> = updateFlow
 }
 
```

Comment:
This affects many updaters, but it seems it's not breaking logic.
I use it to subscribe start updater only once

Reply:
Looks like a valid approach